python/ray/serve/api.py

```diff
@@ -467,6 +509,13 @@ def run(
     Returns:
         RayServeSyncHandle: A handle that can be used to call the application.
     """
+    if host != DEFAULT_HTTP_HOST or port != DEFAULT_HTTP_PORT:
```

Comment:
Non-blocker: If users are explicitly setting their host and/ or port to those default values, then they will not get this warning message. Maybe we can have `None` as default arguments and raise this message if `host` or `port` are not `None`. Then have new if statements to set to the default values if they are `None` so the api continue it's behavior until we deprecated these arguments :)